webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/components/ShowNodeDetailsFromNode.scala

```diff
@@ -143,33 +147,34 @@ class ShowNodeDetailsFromNode(
   }.toBox
 
   val emptyInterval = AgentRunInterval(Some(false), 5, 0, 0, 0) // if everything fails, we fall back to the default entry
-  def getSchedule(nodeInfo: NodeInfo): Box[AgentRunInterval] = {
-    Full(nodeInfo.nodeReportingConfiguration.agentRunInterval.getOrElse(getGlobalSchedule().getOrElse(emptyInterval)))
+  def getSchedule(nodeFact: CoreNodeFact): Box[AgentRunInterval] = {
+    Full(nodeFact.rudderSettings.reportingConfiguration.agentRunInterval.getOrElse(getGlobalSchedule().getOrElse(emptyInterval)))
   }
 
-  def saveSchedule(nodeInfo: NodeInfo)(schedule: AgentRunInterval): Box[Unit] = {
+  def saveSchedule(nodeFact: CoreNodeFact)(schedule: AgentRunInterval): Box[Unit] = {
+    val newNodeFact = nodeFact.modify(_.rudderSettings.reportingConfiguration.agentRunInterval).setTo(Some(schedule))
     val modId       = ModificationId(uuidGen.newUuid)
-    val user        = CurrentUser.actor
-    val newNodeInfo = nodeInfo.copy(
-      nodeInfo.node.copy(nodeReportingConfiguration =
-        nodeInfo.node.nodeReportingConfiguration.copy(agentRunInterval = Some(schedule))
-      )
-    )
+    val cc          = ChangeContext(modId, CurrentUser.actor, DateTime.now(), None, None, CurrentUser.nodePerms)
+
     (for {
-      _ <- nodeRepo.updateNode(newNodeInfo.node, modId, user, None)
+      _ <- nodeFactRepo.save(NodeFact.fromMinimal(newNodeFact))(cc)
```

Comment:
Now us a `ChangeContext` to commit update to Node. 
Two observations:
- perhaps we would like to have moreinfo in `CurrentUser` like user IP, 
- it seems that we want to have a `nodeFactRepo.save` that directly works on `CoreNodeFact` and manage for the us the data conversion (plus it would help it knows how to optimize save)